tests/cnf/core/network/security/tests/nftables-custom-firewall-test.go

```diff
@@ -0,0 +1,505 @@
+	Expect(err).ToNot(HaveOccurred(), "Failed to create FRR test pod")
+
+	return frrPod
+}
+
+func createMasterPodTest(name, nodeName, configmapName string, defaultCMD []string,
+	secondaryNetConfig []*types.NetworkSelectionElement, portNum int) *pod.Builder {
+	frrPod := pod.NewBuilder(APIClient, name, tsparams.TestNamespaceName, NetConfig.FrrImage).
+		DefineOnNode(nodeName).
+		WithTolerationToMaster().
+		WithSecondaryNetwork(secondaryNetConfig).RedefineDefaultCMD(defaultCMD)
+
+	By("Creating FRR master container in the test namespace")
+
+	frrContainer := pod.NewContainerBuilder(
+		containerFrr, NetConfig.CnfNetTestContainer, []string{"/bin/bash", "-c",
+			fmt.Sprintf("testcmd -interface net1 -protocol tcp -port %d -listen", portNum)}).
+		WithSecurityCapabilities([]string{"NET_ADMIN", "NET_RAW", "SYS_ADMIN"}, true)
+
+	frrCtr, err := frrContainer.GetContainerCfg()
+	Expect(err).ToNot(HaveOccurred(), "Failed to get container configuration")
+	frrPod.WithAdditionalContainer(frrCtr).WithLocalVolume(configmapName, "/etc/frr")
+
+	frrPod, err = frrPod.WithPrivilegedFlag().CreateAndWaitUntilRunning(5 * time.Minute)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create FRR test pod")
+
+	return frrPod
+}
+
+func setWorkerNodeListAndLabelForTests(workerNodeList []*nodes.Builder) (map[string]string, []*nodes.Builder) {
+	return NetConfig.WorkerLabelMap, workerNodeList
+}
+
+// DefineBaseConfig defines minimal required FRR configuration.
+func DefineBaseConfig(daemonsConfig, frrConfig, vtyShConfig string) map[string]string {
+	configMapData := make(map[string]string)
+	configMapData["daemons"] = daemonsConfig
+	configMapData["frr.conf"] = frrConfig
+	configMapData["vtysh.conf"] = vtyShConfig
+
+	return configMapData
+}
+
+func removePrefixFromIPList(ipAddressList []string) []string {
+	var ipAddressListWithoutPrefix []string
+	for _, ipaddress := range ipAddressList {
+		ipAddressListWithoutPrefix = append(ipAddressListWithoutPrefix, ipaddr.RemovePrefix(ipaddress))
+	}
+
+	return ipAddressListWithoutPrefix
+}
+
+func buildRoutesMapWithSpecificRoutes(podList []*pod.Builder, nextHopList []string) map[string]string {
+	Expect(len(podList)).ToNot(BeZero(), "Pod list is empty")
+	Expect(len(nextHopList)).ToNot(BeZero(), "Next hop IP addresses list is empty")
+	Expect(len(nextHopList)).To(BeNumerically(">=", len(podList)),
+		fmt.Sprintf("Number of speaker IP addresses[%d] is less than the number of pods[%d]",
+			len(nextHopList), len(podList)))
+
+	routesMap := make(map[string]string)
+
+	for _, frrPod := range podList {
+		if frrPod.Definition.Spec.NodeName == workerNodeList[0].Definition.Name {
+			routesMap[frrPod.Definition.Spec.NodeName] = nextHopList[1]
+		} else {
+			routesMap[frrPod.Definition.Spec.NodeName] = nextHopList[0]
+		}
+	}
+
+	return routesMap
+}
+
+func setStaticRoute(frrPod *pod.Builder, action, destIP string, nextHopMap map[string]string) (string, error) {
```

Comment:
Do we have similar func in metallb tests? if yes pls reuse it

Reply:
There is one in frr. Should I move it up to network<internal?

Reply:
moved